pandas/tseries/offsets.py

```diff
@@ -2487,6 +2488,9 @@ def generate_range(start=None, end=None, periods=None,
 
     """
     if time_rule is not None:
```

Comment:
did you need to change *any* existing tests to account for this?

Reply:
yes, tests.indexes.datetimes.test_date_range test_generate and test_generate_cday

Reply:
is this sufficiently internal that we don't need to do a deprecation cycle?

Reply:
yeah, let's just drop it entirely (fix the tests). I don't think ``generate_range`` is exposed, though maybe add in a note in api breaking changes just to cover.

Reply:
Will do.  OK with moving this to arrays.datetimes while we're at it?